Functions/QualityControl/xASL_qc_GenerateReport.m

```diff
@@ -0,0 +1,418 @@
+    [~] = xASL_sub_parseJson(json, x, primaryFig, [0 0.93 1 0], settings);
+
+    PrintFile = ['xASL_Report_', json.pageIdentifier];
+    PrintPath = fullfile(x.dir.xASLDerivatives, x.SUBJECT, PrintFile); 
+
+    print(primaryFig, PrintPath, '-dpdf', '-bestfit');
+
+    clear primaryFig ax
+end
+
+function xASL_sub_printBlock(input, x, pageFig, settings)
+    position = xASL_str2num(input.position);
+    size = xASL_str2num(input.size);
+    [settings.canvas, line] = xASL_sub_createNewCanvas(position, size, settings.canvas);
+    xASL_sub_parseJson(input, x, pageFig, line, settings);
+end
+
+function line = xASL_sub_PrintPatient(x, figure, line, settings)
+    %Check if participant data exists.
+    ExistsParticipants = xASL_adm_GetFileList(x.D.ROOT, 'participants.tsv');
+    PatientInfo = {};
+
+    if isempty(ExistsParticipants)
+        return
+    end
+
+    sParticipants = xASL_tsvRead(ExistsParticipants{1});
+    nParticipants = size(sParticipants, 1);
+    for iPar=2:nParticipants
+        if sParticipants{iPar, 1} == x.SUBJECT
+            %Extract Participants table, and participants info
+            PatientInfo = [sParticipants(1,:); sParticipants(iPar, :)];
+        end
+    end
+
+    if isempty(PatientInfo)
+        fprintf ('no patient info found to be printed \n');
+        return
+    end
+    
+    line = xASL_sub_PrintText('Participant Information', figure, line, settings );
+    for iEntry=1:size(PatientInfo,2)
+        if strcmp(PatientInfo{1, iEntry},'participant_id')
+            line = xASL_sub_PrintText(['Participant: ', PatientInfo{2, iEntry}], figure, line, settings );
+        elseif strcmp(PatientInfo{1, iEntry},'age')
+            line = xASL_sub_PrintText(['Participant age: ', num2str(PatientInfo{2, iEntry})], figure, line, settings );
+        elseif strcmp(PatientInfo{1, iEntry},'sex')
+            line = xASL_sub_PrintText(['Participant sex: ', PatientInfo{2, iEntry}], figure, line, settings );
+        end
+    end
+
+end
+
+function line = xASL_sub_PrintQC(json, x, figure, line, settings)
+    % Stop if module and field don't exists in output
+    if ~isfield(x.Output, (json.module)) && ~isfield(x.Output.(json.module), json.parameter)
+        return     
+    end
+
+    % Use field specific settings if they exists
+    settings.fontName = 'monospace';
+    if isfield(json, 'settings')
+        settings = xASL_sub_loadSettings(json.settings, settings);
+    end
+
+    TempValue = x.Output.(json.module).(json.parameter);
+    
+    if ~isfield(json, 'alias') 
+        json.alias = json.parameter;
+    end
+
+    if ~isfield(json, 'unit') 
+        json.unit = '';
+    end
+
+    if isfield(json, 'range') 
+        [range] = strsplit(json.range, '-');
+        if (TempValue < xASL_str2num(range{1})) || (TempValue > xASL_str2num(range{2}))
+            settings.color = 'r';
+        end
+        json.range = ['(' json.range ')'];
+    else
+        json.range = '';
+    end
+
+    if isnumeric(TempValue)
```

Comment:
You can use xASL_num2str which deals with isnumeric and different formats of numbers